src/index.ts

```diff
@@ -0,0 +1,312 @@
+/*
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { addPath } from '@actions/core';
+import * as exec from '@actions/exec';
+import * as toolCache from '@actions/tool-cache';
+import * as os from 'os';
+import * as path from 'path';
+import { getReleaseURL } from './format-url';
+import * as downloadUtil from './download-util';
+import * as installUtil from './install-util';
+import { getLatestGcloudSDKVersion } from './version-util';
+import { ExecOptions } from '@actions/exec/lib/interfaces';
+
+export { getLatestGcloudSDKVersion };
+
+/**
+ * Checks if gcloud is installed.
+ *
+ * @param version - (Optional) Cloud SDK version.
+ * @returns true if gcloud is found in toolpath.
+ */
+export function isInstalled(version?: string): boolean {
+  let toolPath;
+  if (version) {
+    toolPath = toolCache.find('gcloud', version);
+    return toolPath != undefined && toolPath !== '';
+  } else {
+    toolPath = toolCache.findAllVersions('gcloud');
+    return toolPath.length > 0;
+  }
+}
+
+/**
+ * Returns the correct gcloud command for OS.
+ *
+ * @returns gcloud command.
+ */
+export function getToolCommand(): string {
+  // A workaround for https://github.com/actions/toolkit/issues/229
+  // Currently exec on windows runs as cmd shell.
+  let toolCommand = 'gcloud';
+  if (process.platform == 'win32') {
+    toolCommand = 'gcloud.cmd';
+  }
+  return toolCommand;
+}
+
+/**
+ * Checks if the project Id is set in the gcloud config.
+ *
+ * @returns true is project Id is set.
+ */
+export async function isProjectIdSet(silent = true): Promise<boolean> {
+  let output = '';
+  const stdout = (data: Buffer): void => {
+    output += data.toString();
+  };
+  const stderr = (data: Buffer): void => {
+    output += data.toString();
+  };
+  const options = {
+    listeners: {
+      stdout,
+      stderr,
+    },
+    silent,
+  };
+
```

Comment:
Looks like there a new [execOutput](https://github.com/actions/toolkit/blob/f0b00fd201c7ddf14e1572a10d5fb4577c4bd6a2/packages/exec/src/interfaces.ts#L41) which may simplify this and others

Reply:
It is somewhat undetermined where gcloud will print to (stdout and stderr). This way keeps it easier to combine both outputs.

Reply:
makes sense